src/ucp/rma/rma_send.c

```diff
@@ -207,6 +207,31 @@ ucs_status_ptr_t ucp_put_nb(ucp_ep_h ep, const void *buffer, size_t length,
     return ucp_put_nbx(ep, buffer, length, remote_addr, rkey, &param);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_put_send_short(ucp_ep_h ep, const void *buffer, size_t length,
+                   uint64_t remote_addr, ucp_rkey_h rkey,
+                   const ucp_request_param_t *param)
+{
+    const ucp_rkey_config_t *rkey_config;
+    uct_rkey_t tl_rkey;
+
+    if (ucs_unlikely(param->op_attr_mask & (UCP_OP_ATTR_FIELD_DATATYPE |
```

Comment:
Isn't it `INVALID_PARAM`?

Reply:
it's UCS_ERR_NO_RESOURCE in order to fallback to using UCP request

Reply:
yes, but not `UCP_OP_ATTR_FIELD_DATATYPE`

Reply:
if UCP_OP_ATTR_FIELD_DATATYPE is set, then it may not be contig, so must check datatype explicitly (for example by allocating request flow)
we could do same as 'tag' here - check if datatype==generic as optimization - but seems not needed for now since most users of this API do not set datatype field at all

Reply:
but it is not really supported now..

Reply:
> but it is not really supported now..

Right, but it's not directly related to the code above, here we just disable fast-path short if datatype is specified

Reply:
ok